### PR TITLE
Update TestEZ and fix tests so they work in studio

### PR DESCRIPTION
**Pull request by oltrep**
TestEZ submodule has been updated to latest.

The usage of `CoreGui` was throwing permission errors when running tests within Studio so I've replaced it with a Folder instance in ReplicatedStorage. Since I've updated TestEZ, I was able to use `beforeEach` and `afterEach` to clean up that instance.

Checklist before submitting:
* [x] ~Added entry to `CHANGELOG.md`~
* [x] Added/updated relevant tests
* [x] ~Added/updated documentation~

### CODE REVIEW

File: bin/run-tests.server.lua [hunk 1]
```diff
@@ -12,14 +12,15 @@ Roact.setGlobalConfig({
...
+local statusCode = (results.failureCount == 0 and #results.errors == 0) and 0 or 1
 
 if __LEMUR__ then
-	if results.failureCount > 0 then
-		os.exit(statusCode)
-	end
+	os.exit(statusCode)
 elseif isRobloxCli then
-	ProcessService:Exit(statusCode)
+	ProcessService:ExitAsync(statusCode)
```
Comment (hmallen99) [inline bin/run-tests.server.lua:25]: What's the reason for this change? Just for my own learning.

File: place.project.json [hunk 2]
```diff
@@ -11,7 +11,7 @@
       },
 
       "TestEZ": {
-        "$path": "modules/testez/lib"
+        "$path": "modules/testez"
```
Comment (hmallen99) [inline place.project.json:14]: Why does this not need to path to modules/testez/src, like in bin/spec.lua? Does it just automatically load the src?

**Review by hmallen99 — APPROVED** [review]
Looks good to me, all the tests still seem to be working. I had a couple questions, but nothing blocking

File: bin/run-tests.server.lua (hunk 1 above)
Comment (oltrep) [reply bin/run-tests.server.lua:25]: At some point the API just changed 😅 Blocking functions are usually marked with an `Async` suffix that's why it got corrected

File: place.project.json (hunk 2 above)
Comment (oltrep) [reply place.project.json:14]: When you give a path to a directory to Rojo, it will look for a file named `default.project.json` and use that to map the directory to instances. If it does not exist, it'll default to just the regular file to instance mapping rules.

I think this file was probably written before Rojo was able to do that
